frontend/src/metabase/query_builder/actions/querying.js

```diff
@@ -93,6 +93,18 @@ export const runQuestionQuery = ({
       : getQuestion(getState());
     const originalQuestion = getOriginalQuestion(getState());
 
+    const queryResults = getQueryResults(getState());
+    const isDirty = getIsDirty(getState());
+
+    const areLatestQueryResultsLoaded =
```

Comment:
For this approach to work I think we will need to introduce a `force` config option -- there are use cases where we expect the query to be rerun even though we have results & the results are not dirty, such as when the user clicks the rerun/refresh button. Not sure if there are other scenarios besides that.